### ISSUE TITLE
Test optimizers segv

### ISSUE THREAD

**Issue by jbmouret** [resibots/limbo#50]
```
➜  limbo git:(libcmaes) ✗ build/src/tests/test_optimizers 
Running 6 test cases...
unknown location:0: fatal error in "test_grid_search_mono_dim": signal: SIGSEGV, si_code: 0 (memory access violation at address: 0x00000000)
../src/tests/test_optimizers.cpp:115: last checkpoint

*** 1 failure detected in test suite "optimisers"

```

**Comment by costashatz** [comment]
I am not getting this seg-fault.

This was on what branch? **libcmaes**? Have you tried on another branch?

**Comment by jbmouret** [comment]
I get it on all the current branches. I tested on OSX/clang.

```bash
➜  limbo git:(remove_boost_shared_ptr) ✗ build/src/tests/test_optimizers 
Running 6 test cases...
unknown location:0: fatal error in "test_grid_search_mono_dim": signal: SIGSEGV, si_code: 0 (memory access violation at address: 0x00000000)
../src/tests/test_optimizers.cpp:115: last checkpoint

*** 1 failure detected in test suite "optimizers"
```

**Comment by costashatz** [comment]
I compiled it with clang too. Do you have output of valgrind or gdb?

**Comment by costashatz** [comment]
This was fixed in #56 !
